src/poller.ts

```diff
@@ -8,82 +8,62 @@ import {
-import { API_URL } from './config';
+import { API_URL, NUM_DIP_ATOMS_PER_TOKEN } from './config';
 
-export class Poller {
+class Poller {
   cluster: string;
 
   callback: (deposit: DIPDeposit) => void;
 
-  splToken: string;
+  splTokenName: string;
 
-  premiumAsset: string;
+  premiumAssetName: string;
 
   expirationSec: number;
 
-  strike: number;
+  strikeTokens: number;
 
-  type: string;
+  callOrPut: string;
 
   constructor(
     cluster: string,
-    splToken: string,
-    premiumAsset: string,
+    splTokenName: string,
+    premiumAssetName: string,
     expirationSec: number,
-    strike: number,
-    type: string,
+    strikeTokens: number,
+    callOrPut: string,
     callback: (deposit: DIPDeposit) => void,
   ) {
     this.cluster = cluster;
     this.callback = callback;
-    this.splToken = splToken;
-    this.premiumAsset = premiumAsset;
+    this.splTokenName = splTokenName;
+    this.premiumAssetName = premiumAssetName;
     this.expirationSec = expirationSec;
-    this.strike = strike;
-    this.type = type;
+    this.strikeTokens = strikeTokens;
+    this.callOrPut = callOrPut;
   }
 
   subscribe(address: string): void {
     console.log('Listening at:', address);
-    // TODO: Use a serum API to get the decimals once at the start, but also
-    // debug why SOL has 8 decimals instead of expected 9.
-    
     const connection = new Connection(API_URL, 'processed' as Commitment);
 
     const callback: AccountChangeCallback = (
       accountInfo: solanaAccountInfo<Buffer>,
       _context: Context,
     ) => {
-      const new_amount: number = u64.fromBuffer(AccountLayout.decode(accountInfo.data).amount).toNumber();
-      let decimals = 6;
-      switch (this.splToken) {
-        // BTC
-        case 'JDXktC6gbDXq4zuW3BT6ToSE7timShHQBL449ULDdoMv':
-        case '9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E':
-          decimals = 6;
-          break;
-        // ETH
-        case 'Hccuen6RkUgEvyL9oSXW8ai9QiQaAiL8ESaqjp9oymBf':
-        case '2FPyTwcZLUg1MDrwsyoP4D6s1tM7hAkHYRjkNb5w6Pxk':
-          decimals = 6;
-          break;
-        // SOL
-        case 'So11111111111111111111111111111111111111112':
-          decimals = 8;
-          break;
-        default:
-          break;
-      }
+      const newAmount: number = u64.fromBuffer(
+        AccountLayout.decode(accountInfo.data).amount,
+      ).toNumber();
 
-      const dip_deposit: DIPDeposit = {
-        splTokenMint: this.splToken,
-        premiumAssetName: this.premiumAsset,
+      const dipDeposit: DIPDeposit = {
+        splTokenName: this.splTokenName,
+        premiumAssetName: this.premiumAssetName,
         expirationMs: this.expirationSec * 1_000,
-        strikeUsdcPerToken: this.strike,
-        callOrPut: this.type,
-        qtyTokens: new_amount / 10 ** decimals,
+        strikeUsdcPerToken: this.strikeTokens,
+        callOrPut: this.callOrPut,
+        qtyTokens: newAmount / 10 ** NUM_DIP_ATOMS_PER_TOKEN,
```

Comment:
Won't this break on SOL?

Reply:
No. code wasnt doing anything. this.splTokenName is a name, not an address.

Reply:
Don't we still need it corrected then? wSOL has 9 decimals.. what am I missing?

Reply:
We are watching the option address which is always 6 decimals, not the vault token account